docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+        """Construct a plugin."""
+        self.provider = provider
+
+        self.MODE = "DATA"
+
+        # The location of the raw data
+        self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]
+        # The location to store training datasets for our model
+        self.train_dir = self.provider.tool_config["datasetConfig"]["trainingSetDir"]
+        # Test datasets, like the training above.
+        self.test_dir = self.provider.tool_config["datasetConfig"]["testSetDir"]
+        # We pull validation datasets from the same location as our training here
+        self.val_dir = self.train_dir
+
+        self.provider.io.info("Plugin initialized.")
+```
+
+Where:
+
+`self.MODE = "DATA"`
+ is our tool's "run mode," as described earlier in [link to section].
+
+And:
+
+ `self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]` 
+
+ And the lines below it utilize the `tool_config` dict (link to tool_config in docs) to access data and tool configuration sent by the UI SDK.
+
+You may have already noticed, but `tool_config` will contain the values we said we'd mock before deploying to Designer.
+Once we have developed our frontend, a user can populate these fields.
+Until then, you can safely use constants for testing. (better phrasing?)
+
+Next, add the following to the `on_complete` method:
+
+```python
+try:
+  if self.MODE == 'DATA':
+    self.setup_data()
+except Exception as e:
+  logger = logger.error(f"{self.on_complete}: repr(e)")
+  self.provider.io.error(f"{self.on_complete}: an error occuried, please see PythonSDK.log for details!")
+```
+#### Supplemental: Error Handling in the SDK
+
+There are two crucial pieces to note here. If you already understand the concepts below, feel free to skip to the next section!
+
+First, the first Python sdk tool runs as a separate (python) process from Designer's process.
+While this provides many benefits (such as arbitrary Python tool requirements like TensorFlow here), this means *designer* cannot directly access the exceptions the Python process throws.
+The SDK does its best to capture generic errors, but the scope of edge cases is impossibly large. 
+It is possible to try and catch the "wrong" error type in Python and unintentionally let an exception pass silently. (See Python docs for details)
+As such, we have seen user cases where tricky exceptions slip off the stack uncaught.
+
+A simple example:
+
+```python
+# where bar = {"key_0": [0, 1]}
+try:
+  foo = bar[key_variable][2]
+except KeyError as e:
+  logger.error(f"An error occured: \n{repr(e)}")
+```
+
+This code will catch _an_ error; here, a `TypeError`. However, it will _not_ run our `except` we intend. Here, a couple more lines of code will go a long way for debugging.
+Your understanding of how the below items function would be best so you can efficiently debug your tool.
+If unsure, check out the [official Python docs](https://docs.python.org/3.8/tutorial/errors.html).
```

Comment:
TW review:

This code catches _an_ error, a `TypeError`. However, it does _not_ run the `except` we intend. Here, a couple more lines of code go a long way for debugging. Your understanding of how the below items function will help you efficiently debug your tool. If unsure, check out the [official Python docs](https://docs.python.org/3.8/tutorial/errors.html).

Reply:
good catch!